### ISSUE TITLE
hocr-pdf output failed

### ISSUE THREAD

**Issue by dinosauria123** [tmbdev/hocr-tools#69]
Thank you for update of hocr-pdf.

I try to convert image to searchable pdf by using hocr-pdf and my gcv2hocr.
hocr-pdf sometimes fails to convert pdf.

Attached scan0002.jpg and scan0002.hocr can convert to pdf.

But attached scan0003.jpg and scan0003.hocr fails to convert to pdf. 
The error messages are below.

> Traceback (most recent call last):
  File "hocr-pdf", line 139, in <module>
    export_pdf(sys.argv[1], 300)
  File "hocr-pdf", line 64, in export_pdf
    add_text_layer(pdf, image, height, dpi)
  File "hocr-pdf", line 73, in add_text_layer
    hocr = etree.parse(hocrfile, html.XHTMLParser())
  File "src/lxml/lxml.etree.pyx", line 3427, in lxml.etree.parse (src/lxml/lxml.etree.c:85131)
  File "src/lxml/parser.pxi", line 1782, in lxml.etree._parseDocument (src/lxml/lxml.etree.c:124005)
  File "src/lxml/parser.pxi", line 1808, in lxml.etree._parseDocumentFromURL (src/lxml/lxml.etree.c:124374)
  File "src/lxml/parser.pxi", line 1712, in lxml.etree._parseDocFromFile (src/lxml/lxml.etree.c:123169)
  File "src/lxml/parser.pxi", line 1115, in lxml.etree._BaseParser._parseDocFromFile (src/lxml/lxml.etree.c:117533)
  File "src/lxml/parser.pxi", line 573, in lxml.etree._ParserContext._handleParseResultDoc (src/lxml/lxml.etree.c:110510
)
  File "src/lxml/parser.pxi", line 683, in lxml.etree._handleParseResult (src/lxml/lxml.etree.c:112276)
  File "src/lxml/parser.pxi", line 613, in lxml.etree._raiseParseError (src/lxml/lxml.etree.c:111124)
lxml.etree.XMLSyntaxError: xmlParseEntityRef: no name, line 469, column 234
>

I can not distinguish the problem comes from hocr-pdf or my gcv2hocr.
If you find problems in my gcv2hocr, please notify me.

[scan0002.hocr.txt](https://github.com/tmbdev/hocr-tools/files/481893/scan0002.hocr.txt)
[scan0002.jpg](https://cloud.githubusercontent.com/assets/3067670/18661903/db923192-7f52-11e6-8eb7-49dbf4fd9897.jpg)
[scan0003.jpg](https://cloud.githubusercontent.com/assets/3067670/18661902/db8e888a-7f52-11e6-8de7-45b82b21528a.jpg)
[scan0003.hocr.txt](https://github.com/tmbdev/hocr-tools/files/481892/scan0003.hocr.txt)

**Comment by dinosauria123** [comment]
Sorry, I found my hocr file.

"&" character is remain in output of scan.003.hocr at line 469.

I will fix my gcv2hocr.

**Comment by zuphilip** [comment]
Okay, good to hear that you found the error and can fix it. 

Can you confirm that the fixed file runs through `horc-pdf`?

**Comment by dinosauria123** [comment]
Thank you for your comment.

This will help to fix my gcv2hocr.

http://stackoverflow.com/questions/12524908/how-to-escape-in-xml

I will check output of PDF file after hocr output is fixed.

I scanned old Japanese camera book and I will covert it to searchable PDF file.
It has 70 jpeg files may help check our code work well or not.

**Comment by dinosauria123** [comment]
I have committed out to fix xml escape characters problem my gcv2hocr at GitHub.

Now, hocr-pdf works fine and I can make searchable pdf file from 70 jpeg images from old Japanese camera book.

I made a simple shell script to extract text from image by Google Cloud Vision OCR and make searchable pdf from them.

>#!/bin/sh
a=1
while [ $a -le 100 ]
do
    sh ./gcvocr.sh scan00$a.jpg "Your API Code"
    echo "Google OCR scan00$a.jpg"
    ./a.out scan00$a.jpg.json scan00$a.hocr
    echo "convert scan00$a.hocr"
    a=`expr $a + 1`
done
python hocr-pdf . > output.pdf

If you are interested in my output pdf, download it here (about 40MB)
https://t.co/BUjCMgVnyr

It was written in 1928, public domein in Japan.
